### DISCUSSION TITLE
Audit Logging Feature for custom actions

### DISCUSSION THREAD

**Discussion by opya** [jeremyevans/rodauth#338] [General]
Hello @jeremyevans and thank you for the awesome roda, rodauth and sequel gems.

I was wondering if there is a way to use the audit logging feature to log custom actions:

```ruby
audit_log_message_for :product_delete, "Product with id: #{product.id} was deleted by #{account.email}:#{account_id}"
```

**Comment by jeremyevans** [comment]
As long as your custom action uses Rodauth before hooks, the audit_logging feature will automatically support it.  If it does not use Rodauth before hooks, you will have to call `add_audit_log` manually.  In either case, the code you gave for `audit_log_message_for` should work.

**Comment by opya** [comment]
Thank you, Jeremy.
